compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmInstructionGen.java

```diff
@@ -2324,6 +2357,10 @@ void generateInstructions(int localVarOffset, BIRInstruction inst) {
                     generateNegateIns((BIRNonTerminator.UnaryOP) inst);
                     break;
                 case PLATFORM:
+                    if (inst instanceof ObservabilityJMethodCall) {
```

Comment:
Shall we move this check inside the `generatePlatformIns` method, because `ObservabilityJMethodCall` is also a platform instruction?